### PR TITLE
Optimize Dockerfile and add .dockerignore

### PR DESCRIPTION
**Pull request by theinfosecguy**
Changes include:

- Removed the `mkdir` command as the `WORKDIR` instruction automatically creates the folder if it doesn't already exist.

- Added a .dockerignore file to prevent unnecessary files (such as README, tests, Dockerfile itself, etc.) from being included in the Docker context, thereby speeding up the build process.

- Combined two `RUN` instructions into one in order to reduce the number of layers in the final image.

### CODE REVIEW

**Review by another-rex — COMMENTED** [review]
Thanks! The dockerfile updates look good, though I don't think the .dockerignore is necessary, as it might hinder us wanting to run tests in docker in the future, while not significantly speeding up the build process (there's not that many files being ignored).

Can you remove the .dockerignore?

**Comment by theinfosecguy** [comment]
> Can you remove the .dockerignore?

Removed the `.dockerignore` file.

**Review by another-rex — APPROVED** [review]
LGTM! Thanks!

**Comment by theinfosecguy** [comment]
@another-rex - can we merge this?

File: Dockerfile [hunk 1]
```diff
@@ -12,27 +12,21 @@
...
-RUN git config --global --add safe.directory '*'
+RUN apk --no-cache add ca-certificates git && \
+    git config --global --add safe.directory '*'
 
 WORKDIR /root/
-COPY --from=0 /src/osv-scanner ./
+COPY --from=builder /src/osv-scanner .
 
-ENTRYPOINT ["/root/osv-scanner"]
+ENTRYPOINT ["./osv-scanner"]
```
Comment (another-rex) [inline Dockerfile:32]:
```suggestion
ENTRYPOINT ["/root/osv-scanner"]
```

**Pull request merged**